crates/volta-migrate/src/v2.rs

```diff
@@ -0,0 +1,158 @@
+        let platform_file = old.home.default_platform_file();
+        if platform_file.exists() {
+            let platform_json = read_to_string(platform_file).with_context(|_| {
+                ErrorDetails::ReadPlatformError {
+                    file: platform_file.to_owned(),
+                }
+            })?;
+            let mut existing_platform = Platform::from_json(platform_json)?;
+
+            if let Some(ref mut node_version) = &mut existing_platform.node {
+                if let Some(npm) = &node_version.npm {
+                    if *npm == load_default_npm_version(&node_version.runtime)? {
+                        node_version.npm = None;
+                        write(platform_file, existing_platform.into_json()?).with_context(
+                            |_| ErrorDetails::WritePlatformError {
+                                file: platform_file.to_owned(),
+                            },
+                        )?;
+                    }
+                }
+            }
+        }
+
+        // Move node_image_dir Up one directory (V1 -> V2)
+        let temp_dir =
+            tempdir_in(new_home.tmp_dir()).with_context(|_| ErrorDetails::CreateTempDirError {
+                in_dir: new_home.tmp_dir().to_owned(),
+            })?;
+        let node_installs = read_dir_eager(old.home.node_image_root_dir())
+            .with_context(|_| ErrorDetails::ReadDirError {
+                dir: old.home.node_image_root_dir().to_owned(),
+            })?
+            .filter_map(|(entry, metadata)| {
+                if metadata.is_dir() {
+                    parse_version(entry.file_name().to_string_lossy()).ok()
+                } else {
+                    None
+                }
+            });
+
+        for node_version in node_installs {
+            let npm_version = load_default_npm_version(&node_version)?;
+            let old_install = old
+                .home
+                .node_image_dir(&node_version.to_string(), &npm_version.to_string());
+
+            if old_install.exists() {
+                let temp_image = temp_dir.path().join(node_version.to_string());
+                let new_install = new_home.node_image_dir(&node_version.to_string());
+                rename(&old_install, &temp_image).with_context(|_| {
+                    ErrorDetails::SetupToolImageError {
+                        tool: "Node".to_string(),
+                        version: node_version.to_string(),
+                        dir: temp_image.clone(),
+                    }
+                })?;
+                ensure_dir_does_not_exist(&new_install)?;
+                rename(&temp_image, &new_install).with_context(|_| {
+                    ErrorDetails::SetupToolImageError {
+                        tool: "Node".to_string(),
+                        version: node_version.to_string(),
+                        dir: new_install.clone(),
+                    }
+                })?;
+            }
```

Comment:
There's enough going on here that I'd suggest extracting this into a standalone function. I think it'd be a lot clearer to read something like:

```rust
for node_version in node_installs {
  replace_default_npm_installation(&node_version, &npm_version)?;
}
```

—or something like that, anyway, not sure about the best name.

Reply:
Good thinking, that'll definitely make it more grokkable.